apis/s3/src/test/java/org/jclouds/s3/S3ClientTest.java

```diff
@@ -501,6 +504,86 @@ public void testDisableBucketLogging() throws SecurityException, NoSuchMethodExc
+      assertPayloadEquals(request, null, null, false);
+
+      assertResponseParserClassEquals(method, request, BucketConfigurationHandler.class);
+      assertSaxResponseParserClassEquals(method, null);
+      assertFallbackClassEquals(method, null);
+
+      checkFilters(request);
+   }
+
+   public void testGetBucketEncryption() throws SecurityException, NoSuchMethodException, IOException {
+      Invokable<?, ?> method = method(S3Client.class, "getBucketConfiguration", String.class, BucketConfigOptions[].class);
+      GeneratedHttpRequest request = processor.createRequest(method, ImmutableList.<Object> of("bucket", new BucketConfigOptions().encryption()));
+
+      assertRequestLineEquals(request, "GET https://bucket." + url + "/?encryption=true HTTP/1.1");
+      assertNonPayloadHeadersEqual(request, "Host: bucket." + url + "\n");
+      assertPayloadEquals(request, null, null, false);
+
+      assertResponseParserClassEquals(method, request, BucketConfigurationHandler.class);
+      assertSaxResponseParserClassEquals(method, null);
+      assertFallbackClassEquals(method, null);
+
+      checkFilters(request);
+   }
+
+   public void testDeleteBucketConfiguration() throws SecurityException, NoSuchMethodException, IOException {
+      Invokable<?, ?> method = method(S3Client.class, "deleteBucketConfiguration", String.class, BucketConfigOptions[].class);
+      GeneratedHttpRequest request = processor.createRequest(method, ImmutableList.<Object> of("bucket"));
+
+      assertRequestLineEquals(request, "DELETE https://bucket." + url + "/ HTTP/1.1");
+      assertNonPayloadHeadersEqual(request, "Host: bucket." + url + "\n");
+      assertPayloadEquals(request, null, null, false);
+
+      assertResponseParserClassEquals(method, request, BucketConfigurationHandler.class);
+      assertSaxResponseParserClassEquals(method, null);
+      assertFallbackClassEquals(method, MapHttp4xxCodesToExceptions.class);
+
+      checkFilters(request);
+   }
+
+   public void testPutBucketConfiguration() throws SecurityException, NoSuchMethodException, IOException {
+
+      Invokable<?, ?> method = method(S3Client.class, "putBucketConfiguration", String.class, S3Object.class,
+              BucketConfigOptions[].class);
+      GeneratedHttpRequest request = processor.createRequest(method, ImmutableList.<Object> of("bucket",
+              blobToS3Object.apply(BindBlobToMultipartFormTest.TEST_BLOB), new BucketConfigOptions().lifecycle()));
+
+      assertRequestLineEquals(request, "PUT https://bucket." + url + "/?lifecycle=true HTTP/1.1");
+      assertNonPayloadHeadersEqual(request, "Host: bucket." + url + "\n");
+      assertPayloadEquals(request, "hello", "text/plain",
+              false);
+
+      assertResponseParserClassEquals(method, request, BucketConfigurationHandler.class);
+      assertSaxResponseParserClassEquals(method, null);
+      assertFallbackClassEquals(method, null);
+
+      checkFilters(request);
```

Comment:
Can you add a live test for adding multiple objects with the same name and different versions?  These unit tests are good but comparing against live AWS will give me more confidence about this PR.